Sources/Plasma/CoreLib/hsStream.h

```diff
@@ -103,57 +103,42 @@ enum {
+    void            ReadLE16(size_t count, uint16_t values[]);
     uint32_t        ReadLE32();
-    void            ReadLE32(int count, uint32_t values[]);
+    void            ReadLE32(size_t count, uint32_t values[]);
 
     void            WriteBOOL(bool value);
     void            WriteBool(bool value);
 
     void            WriteByte(uint8_t value);
     void            WriteLE16(uint16_t value);
-    void            WriteLE16(int count, const uint16_t values[]);
+    void            WriteLE16(size_t count, const uint16_t values[]);
     void            WriteLE32(uint32_t value);
-    void            WriteLE32(int count, const  uint32_t values[]);
-
-
-    /* Overloaded  Begin (8 & 16 & 32 int)*/
-    /* yes, swapping an 8 bit value does nothing, just useful*/
-    void            ReadLE(bool* value) { *value = this->ReadByte() ? true : false; }
-    void            ReadLE(uint8_t* value) { *value = this->ReadByte(); }
-    void            ReadLE(uint16_t* value) { *value = this->ReadLE16(); }
-    void            ReadLE(uint32_t* value) { *value = this->ReadLE32(); }
-    void            WriteLE(bool value) { this->Write(1,&value); }
-    void            WriteLE(uint8_t value) { this->Write(1,&value); }
-    void            WriteLE(uint16_t value) { this->WriteLE16(value); }
-    void            WriteLE(uint32_t value) { this->WriteLE32(value); }
-    void            ReadLE(int8_t* value) { *value = this->ReadByte(); }
-    void            ReadLE(char* value) { *value = (char)this->ReadByte(); }
-    void            ReadLE(int16_t* value) { *value = (int16_t)this->ReadLE16(); }
-    void            ReadLE(int32_t* value) { *value = (int32_t)this->ReadLE32(); }
-    void            WriteLE(int8_t value) { this->Write(1,&value); }
-    void            WriteLE(char value) { this->Write(1,(uint8_t*)&value); }
-    void            WriteLE(int16_t value) { this->WriteLE16((uint16_t)value); }
-    void            WriteLE(int32_t value) { this->WriteLE32((uint32_t)value); }
-    /* Overloaded  End */
-
+    void            WriteLE32(size_t count, const uint32_t values[]);
 
     float           ReadLEFloat();
-    void            ReadLEFloat(int count, float values[]);
+    void            ReadLEFloat(size_t count, float values[]);
     double          ReadLEDouble();
-    void            ReadLEDouble(int count, double values[]);
+    void            ReadLEDouble(size_t count, double values[]);
+
     void            WriteLEFloat(float value);
-    void            WriteLEFloat(int count, const float values[]);
+    void            WriteLEFloat(size_t count, const float values[]);
     void            WriteLEDouble(double value);
-    void            WriteLEDouble(int count, const double values[]);
+    void            WriteLEDouble(size_t count, const double values[]);
 
+    template <typename T>
+    inline void ReadLE(T* value)
+    {
+        Read(sizeof(T), value);
+        *value = hsToLE(*value);
+    }
```

Comment:
Could we delete potentially bugprone instances of this, eg `ReadLE<size_t>` and `ReadLE<hsSsize_t>`?

Reply:
Not really, since `size_t` may alias `uint32_t` on some environments (for example) :/